util/ifdtool/ifdtool.c

```diff
@@ -1506,17 +1506,219 @@ static void set_pchstrap(struct fpsba *fpsba, const struct fdbar *fdb, const int
 	fpsba->pchstrp[strap] = value;
 }
 
+static int ifd2_platform_get_hap_location(struct fpsba *fpsba)
+{
+	switch (platform) {
+	case PLATFORM_SKLKBL:
+		/* No exceptions, always strap 0 */
+		return 0;
+	case PLATFORM_IFD2:
+		printf("To change the AltMeDisable/HAP bit please provide the platform as parameter\n");
+		return -1;
+	case PLATFORM_CNL:
+		/*
+		 * This gets little tricky here.
+		 * 1. CNL/CFL/CML-LP strap 106 (FPSBA + 0x70), but
```

Comment:
Would it not make more sense to pass the platform name / variant / SKU as an argument to ifdtool at this point?

Reply:
I didn't want to redesign ifdtool that much. Besides it is for dumping and modifying FD in scope of region boundaries, flash master permissions etc. Having support for HAP is an extra nice-to-have.